Adding rag evaluation application (#489)
NVIDIA-AI-Blueprints/rag: skill-source/.agents/skills/rag-blueprint/references/configure/evaluation.md

# Evaluation

## When to Use

> - The user wants to measure RAG pipeline quality.
> 
> - User asks about accuracy, relevancy, groundedness, or recall metrics.
> 
> - The user wants to run the filesystem benchmark evaluator (`scripts/eval/evaluate_rag.py`) with corpus/ plus `train.json`.

Reviewer: ```suggestion
- The user wants to evaluate the quality of a RAG pipeline.

- The user is asking about metrics such as accuracy, relevance, groundedness, or recall.

- The user intends to run the filesystem benchmark evaluator (scripts/eval/evaluate_rag.py) using corpus/ and `train.json`.
```

## When to Use
- User wants to measure RAG pipeline quality
- User asks about accuracy, relevancy, groundedness, or recall metrics

> - User wants to run the **filesystem benchmark** evaluator (`scripts/eval/evaluate_rag.py`) with `corpus/` + `train.json`

Reviewer: ```suggestion
- The user wants to measure RAG pipeline quality.

- The user is asking about accuracy, relevancy, groundedness, or recall metrics.

- The user wants to run the filesystem benchmark evaluator (`scripts/eval/evaluate_rag.py`) with corpus/ plus `train.json`.
```

## Process
1. Read `docs/evaluate.md` for full evaluation methodology and setup
2. Choose the path:

>    - **Notebooks** — interactive RAGAS / recall workflows against a running stack

Reviewer: ```suggestion
   - Notebooks — interactive RAGAS / recall workflows against a running stack
```

>    - **CLI benchmark** — on-disk datasets and `evaluate_rag.py` (see skill `rag-eval` and `scripts/eval/README.md`)

Reviewer: ```suggestion
   - CLI benchmark — on-disk datasets and `evaluate_rag.py` (see skill `rag-eval` and `scripts/eval/README.md`)
```

3. Run evaluation against the deployed RAG pipeline

## Agent-Specific Notes
- **Notebooks** use the RAGAS framework; `evaluation_01_ragas.ipynb` covers answer accuracy, context relevancy, and groundedness
- **Recall@k** (top-k cutoffs) is covered in `evaluation_02_recall.ipynb` only — it is **not** wired into `scripts/eval/evaluate_rag.py` (no `train_extended.json` or recall sidecar in that driver)

> - **`evaluate_rag.py`** ingests `corpus/`, queries `/v1/generate`, then runs RAGAS judge metrics via `rag_evaluator/`; requires `NVIDIA_API_KEY` and matches current CLI in that script

Reviewer: ```suggestion
- Notebooks use the RAGAS framework; `evaluation_01_ragas.ipynb` covers answer accuracy, context relevancy, and groundedness
- Recall@k (top-k cutoffs) is covered in `evaluation_02_recall.ipynb` only — it is not wired into `scripts/eval/evaluate_rag.py` (no `train_extended.json` or recall sidecar in that driver)
- `evaluate_rag.py` ingests `corpus/`, queries `/v1/generate`, then runs RAGAS judge metrics via `rag_evaluator/`; requires `NVIDIA_API_KEY` and matches current CLI in that script
```
